agent/app/agent_unix.go

```diff
@@ -0,0 +1,76 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	log "github.com/cihub/seelog"
+	"golang.org/x/net/context"
+)
+
+// startSigchldHandler registers a channel to receiev SIGCHLD signals for
+// the agent process. On receiving SIGCHLD, it invokes the wait4 syscall
+// so that child processes are appropriately cleaned up.
+func (agent *ecsAgent) startSigchldHandler(ctx context.Context) {
+	signals := make(chan os.Signal)
```

Comment:
You'll want to specify a buffer size here.  From the [godoc](https://golang.org/pkg/os/signal/#Notify):

> Package signal will not block sending to c: the caller must ensure that c has sufficient buffer space to keep up with the expected signal rate. For a channel used for notification of just one signal value, a buffer of size 1 is sufficient.